### ISSUE TITLE
error loading config: field writeNewSeriesLimitPerSecond not found in type config.DBConfiguration

### ISSUE THREAD

**Issue by skupjoe** [m3db/m3db-operator#295]
I am attempting to deploy an m3db cluster to k8s (AWS + persistent volumes) for the using operator, via:
`kubectl apply -f ./m3db-cluster.yaml`

It deploys the m3db cluster fine, but the pods don't start and the pod logs for each StatefulSet shows the error:
```
error loading config: unable to load config from [/etc/m3db/m3.yml]: yaml: unmarshal errors:
  line 72: field writeNewSeriesLimitPerSecond not found in type config.DBConfiguration
```
![image](https://user-images.githubusercontent.com/5000412/125872729-70c02f1f-ab40-4281-b51b-b87f90ee3cc2.png)

To fix the issue, I go to the cluster configMap and remove the line matching `writeNewSeriesLimitPerSecond` and recreate the pods:
![image](https://user-images.githubusercontent.com/5000412/125872786-d5ca848f-8180-4cae-aaa1-d243afc38040.png)

Operator version:
`quay.io/m3db/m3db-operator:v0.12.0 (latest)`

m3dbnode version:
`quay.io/m3db/m3dbnode:v1.1.0 (255a4b14d)`

etcd version:
`quay.io/coreos/etcd:v3.5.0 (latest)`

Kubernetes version:
`GitVersion:"v1.18.16-eks-7737de", GitCommit:"7737de131e58a68dda49cdd0ad821b4cb3665ae8"`

**Comment by gibbscullen** [comment]
@skupjoe looking into this.. where did you find the configuration you're currently using? Also could be helpful to take a look at our doc on creating an m3 cluster with k8s https://m3db.io/docs/cluster/kubernetes_cluster/.

**Comment by skupjoe** [comment]
Hi @gibbscullen. In this case, I was following [the operator docs](https://m3db.io/docs/operator/getting_started/create_cluster) directly, and the configMap that supplies the `m3.yml` configuration gets created automatically. So, I am using the default-provided configuration. I [installed](https://m3db.io/docs/operator/getting_started/installation/#helm) the operator from the latest stable Helm chart.

I used [this](https://gist.github.com/skupjoe/47b0f4c304a82f40b1a59e788ba2d544) M3DBCluster manifest\-\-a slight modification on the [persistent cluster template](https://m3db.io/docs/operator/getting_started/create_cluster/#m3db) in the doc steps (I tried using both master & latest images)\-\-and I ran `kubectl apply -f ./m3db-cluster.yaml` which then automatically launched all the resources and made the `m3.yml` configMap which has this bug.

In order to fully overcome this error, I needed to follow [these instructions](https://m3db.io/docs/operator/configuration/configuring_m3db) and create/apply my own [configMap](https://gist.github.com/skupjoe/f1266eb7fb94cf1f935cf2c35c4cac8f) (exactly the same as the default, minus the line with `writeNewSeriesLimitPerSecond`) and specify it in my `m3-cluster.yaml` [manifest](https://gist.github.com/skupjoe/32f98323d3e408f32397c0e2fbe6d552) and apply this instead.

**Comment by wesleyk** [comment]
ah @skupjoe apologies for the misconfiguration! It looks like the operator docs are pointing to an older version of the operator (v0.10.0) when it should be pointing to the latest version (v0.13.0). I'll go ahead and update that now!

**Comment by wesleyk** [comment]
Here's the docs change: https://github.com/m3db/m3/pull/3621. We've also published the helm charts for v0.13.0, so future installations should get that version!
